pkg/sources/chunker_test.go

```diff
@@ -137,22 +145,24 @@ func TestNewChunkedReader(t *testing.T) {
 			input:      strings.Repeat("a", 2048),
 			chunkSize:  1024,
 			peekSize:   1024,
-			wantChunks: []string{strings.Repeat("a", 1024), strings.Repeat("a", 1024)},
+			totalSize:  1024 + 1024,
+			wantChunks: []string{strings.Repeat("a", 2048), strings.Repeat("a", 1024)},
 			wantErr:    false,
 		},
 		{
 			name:       "Larger than totalSize",
 			input:      strings.Repeat("a", 4096),
 			chunkSize:  1024,
 			peekSize:   1024,
-			wantChunks: []string{strings.Repeat("a", 1024), strings.Repeat("a", 1024), strings.Repeat("a", 1024), strings.Repeat("a", 1024)},
+			totalSize:  1024 + 1024,
+			wantChunks: []string{strings.Repeat("a", 2048), strings.Repeat("a", 2048), strings.Repeat("a", 2048), strings.Repeat("a", 1024)},
 			wantErr:    false,
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			readerFunc := NewChunkReader(WithChunkSize(tt.chunkSize), WithPeekSize(tt.peekSize))
+			readerFunc := NewChunkReader(WithChunkSize(tt.chunkSize), WithPeekSize(tt.peekSize), WithTotalChunkSize(tt.totalSize))
```

Comment:
It seems kind of odd to me having `WithTotalChunkSize` option. Isn't the total chunk size a function of the chunk size and peek size (namely `ChunkSize + PeekSize`)?